### ISSUE TITLE
DGMR Model Code Availability

### ISSUE THREAD

**Issue by jacobbieker** [deepmind/deepmind-research#290]
Is there any plans to release the model code for the nowcasting DGMR?

**Comment by shmurai** [comment]
FYI: the pseudocode seems to be released in `gs://dm-nowcasting-example-data/pseudocode.zip` (in the same bucket as the example datasets).

**Comment by jacobbieker** [comment]
Awesome! Thanks for finding that. Looks quite complete, only thing I notice is that `layers.txt` and `latent_stack.txt` are the  same content, so there might be a mistake on copy/paste or something, but that does help a ton!

**Comment by TQRTQ** [comment]
The code is great and can help me understand the details of the paper better, but I found that there are many codes missing in layers.txt, such as SNConv3d and ApplyAlongAxis. I want to reproduce this work, can you give me the complete code，please.

**Comment by TQRTQ** [comment]
The code is great, I can understand the details of the paper better, but I found a lot of code missing in layer.txt, such as SNConv3d, ApplyAlongAxis.
I have finished a Pytorch version of DGMR code and uploaded it to my [GitHub,](https://github.com/TQRTQ/DGMR) after looking at the pseudo code I found some details that I misunderstood. 
I want to reproduce this work, can you give me the complete code？please！

**Comment by bugsuse** [comment]
> Awesome! Thanks for finding that. Looks quite complete, only thing I notice is that `layers.txt` and `latent_stack.txt` are the same content, so there might be a mistake on copy/paste or something, but that does help a ton!

The pseudocode.zip has been updated, and this issue is fixed.

**Comment by jacobbieker** [comment]
Thanks for pointing it out! That helps a lot

**Comment by hyungting** [comment]
Thanks for updating the pseudo code! I am trying to reproduce this work, and I'm wondering if you can generally provide some training advice for me. I followed most of the settings in the paper except several settings that could shorten training period. The training dataset of mine is just simply filtered radar images in dbz (if 40% grids > dbz-> training set). I shortened the predict lead time to 6 frames and set batch size to 45 (due to the limit of my GPU). However, I always got smoothed predictions when the training epoch reached to 300~400, and the loss of discriminators and generators did not decrease. Can you provide some training tips or share your training experience? Thank you!

My code of DGMR in PyTorch version: https://github.com/hyungting/DGMR-pytorch
This is my training record of DGMR: https://docs.google.com/presentation/d/13PFavy9GC7Gf6T1k9kFr8G9Xsq_Y2t3WCbgoo0Npc_4/edit?usp=sharing
